backend/run/copr-change-storage

```diff
@@ -150,7 +150,7 @@ def main():
                 if not builddir_entry.is_dir():
                     continue
                 builddir = builddir_entry.name
-                build_id = builddir.split('-')[0]
+                build_id = builddir.split('-')[0].lstrip("0")
```

Comment:
Even though, we could use `str(int(...))` pattern to make the script fail in case of unexpected value.

Reply:
This actually caught an error!

```
Uploaded to Pulp: hello-2.8-1.fc41.src.rpm
Uploaded to Pulp: hello-2.8-1.fc41.x86_64.rpm
Uploaded to Pulp: hello-debuginfo-2.8-1.fc41.x86_64.rpm
Uploaded to Pulp: hello-debugsource-2.8-1.fc41.x86_64.rpm
OK: /var/lib/copr/public_html/results/frostyx/test-pulp-20/fedora-41-x86_64/08561800-hello
Traceback (most recent call last):
  File "/opt/copr/backend/run/copr-change-storage", line 217, in <module>
    main()
    ~~~~^^
  File "/opt/copr/backend/run/copr-change-storage", line 153, in main
    build_id = str(int(builddir.split("-")[0]))
                   ~~~^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: invalid literal for int() with base 10: 'repodata'
```

Reply:
It wouldn't be a problem in production because we `if not is_valid_build_directory(builddir)` just below this line and the directory would be skipped but I moved it up.